File: Python/202_Happy Number.py
class Solution:
    def isHappy(self,n):    
        set_num = set()
        while n != 1:
            n = self.getsum(n)
            if n in set_num:
                return False
            set_num.add(n)           
            
        return True 
    
    def getsum(self,n):
        total = 0
        while n:
            total = total + (n%10)**2
            n = n//10
        return total
        
    def isHappy_res(self, n: int) -> bool:
        isExist = set()         
        def helper(num):       
            if num == 1:
                return True  
            elif num in isExist:
                return False
            isExist.add(num)
            num = sum(map(lambda x:int(x)**2,str(num)))
            return helper(num)
        
        return helper(n)

File: JavaScript/202_Happy Number.js

var isHappy = function(n) {
    var map = new Map();
    while (n != 1){
        n =   getSum(n);
        if (map.has(n)) {
               
            return false;  
        }            
        map.set(n, n);                 
    }      
    return true;
    
}

function getSum(n){
    var s = n.toString();
    var sum = 0;
    for (var i = 0 ; i < s.length ; i++){
            sum = sum + parseInt(s[i])*parseInt(s[i]);
    }   
    return sum;
    
}



// var isHappy = function(n) {
//     var sum = 0;
//     var map = new Map();
  
//     while(true) {
//       if (n === 1) {
//         return true;
//       }
//       if (map.has(n)) {
//         return false;
//       }
//       else {
//         map.set(n, n);
//       }
  
//       sum = 0;
//       while(n !== 0) {
//         sum += Math.floor(n % 10) * Math.floor(n % 10);
//         n = Math.floor(n / 10);
//       }
  
//       n = sum;
//     }
//   };



console.log(isHappy(19))